plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreTableColumnManager.java

```diff
@@ -207,9 +207,13 @@ protected void addObjectModifyActions(DBRProgressMonitor monitor, DBCExecutionCo
 //        ALTER [ COLUMN ] column SET STORAGE { PLAIN | EXTERNAL | EXTENDED | MAIN }
         String prefix = "ALTER " + table.getTableTypeName() + " " + DBUtils.getObjectFullName(table, DBPEvaluationContext.DDL) +
             " ALTER COLUMN " + DBUtils.getQuotedIdentifier(column) + " ";
-        String typeClause = column.getFullTypeName();
-        if (column.getDataSource().getServerType().supportsAlterTableColumnWithUSING() && column.getDataType() != null) {
-            typeClause += " USING " + DBUtils.getQuotedIdentifier(column) + "::" + column.getDataType().getName();
+        final PostgreDataType type = column.getDataType();
+        final String fullTypeName = type != null ? DBUtils.getObjectFullName(type, DBPEvaluationContext.DDL) : column.getFullTypeName();
+        String typeClause = fullTypeName;
+        if (column.getDataSource().getServerType().supportsAlterTableColumnWithUSING()) {
+            typeClause += " USING ";
+            typeClause += column.getDataSource().getSQLDialect().getTypeCastClause(column, DBUtils.getQuotedIdentifier(column), true);
```

Comment:
getTypeCastClause already appends `::type-fqn`

Reply:
It doesn't. We need a sausage in a form of `[column_name]::text::[target_type]` for enums and `[column_name]::[target_type]` for other types.

The `getTypeCastClause` appends just the `::text` part for enums and several other types, but we still need to cast to the target type.